src/main/java/com/google/devtools/build/lib/rules/cpp/CcModule.java

```diff
@@ -386,9 +390,38 @@ public LibraryToLink createLibraryLinkerInput(
     Artifact dynamicLibrary = nullIfNone(dynamicLibraryObject, Artifact.class);
     Artifact interfaceLibrary = nullIfNone(interfaceLibraryObject, Artifact.class);
 
-    Artifact notNullArtifactForIdentifier = null;
     StringBuilder extensionErrorsBuilder = new StringBuilder();
     String extensionErrorMessage = "does not have any of the allowed extensions";
+
+    PathFragment dynamicLibraryPathFragment = null;
+    if (dynamicLibraryPath != null && !dynamicLibraryPath.isEmpty()) {
+      dynamicLibraryPathFragment = PathFragment.create(dynamicLibraryPath);
+      if (dynamicLibraryPathFragment.isEmpty() || dynamicLibraryPathFragment.isAbsolute() || dynamicLibraryPathFragment.containsUplevelReferences()) {
+        throw new EvalException(location, String.format("dynamic_library_symlink_path must be a relative file path. Got '%s'", dynamicLibraryPathFragment.toString()));
+      }
+      if (!Link.ONLY_SHARED_LIBRARY_FILETYPES.matches(dynamicLibraryPathFragment.getBaseName())) {
+        String extensions = Link.ONLY_SHARED_LIBRARY_FILETYPES.toString();
+        extensionErrorsBuilder.append(
+            String.format("'%s' %s %s", dynamicLibraryPathFragment.toString(), extensionErrorMessage, extensions));
```

Comment:
There's no need to call toString() on objects within String.format(), as it already does the appropriate conversion
(There are also 3 other instances of this in this file)

Reply:
Good point, I've removed the calls to `.toString()`.